default/AI/ColonisationAI.py

```diff
@@ -593,6 +602,12 @@ def get_colony_fleets():
 
     evaluated_outpost_planets = assign_colonisation_values(evaluated_outpost_planet_ids,
                                                            AIFleetMissionType.FLEET_MISSION_OUTPOST, None, empire)
+    # if outposted planet would be in supply range, let outpost value be best of outpost value or colonization value
+    for pid in set(evaluated_outpost_planets).intersection(evaluated_colony_planets):
+        if planet_supply_cache.get(pid, -99) >=0:
```

Comment:
Yes, it can contain negative values.  I think this is explained with the various contributions, but I can add some comments painting a picture for it at a single spot.  As noted at it's initial assignment, ```planet_supply_cache``` includes the local system_supply number. It is (currently) just this system supply that can make the total negative, there are no current conditions under which the planetary contribution would be negative, but that could change at any time with the addition of a special.  This system supply number can be negative; nonnegative numbers are the same as the standard supply calculation, and negative values indicate the number of jumps to the nearest non-negative supply.  So if this planet supply cache value is non-negative for a planet, then that tells the AI that the planet would be supply connected if it were outposted by the AI (or invaded if it has a current population).